docs/resources/server/enchantments/index.md

```diff
@@ -0,0 +1,322 @@
+    // Acquire the Increment instance from the enchantment holder (or null if this is a different enchantment)
+    Increment increment = enchantmentHolder.value().effects().get(INCREMENT.get());
+
+    // If this enchant has an Increment component, use it.
+    if(increment != null){
+        atomicValue.set(increment.add(atomicValue.get()));
+    }
+});
+
+int modifiedValue = atomicValue.get();
+// Use the now-modified value elsewhere in your game logic.
+```
+
+First, we invoke one of the overloads of `EnchantmentHelper#runIterationOnItem`. This function accepts an `EnchantmentHelper.EnchantmentVisitor`, which is a functional interface that accepts an enchantment and its level, and is invoked on all of the enchantments that the given itemstack has (essentially a `BiConsumer<Enchantment, Integer>`).
+
+To actually perform the adjustment, use the provided `Increment#add` method. Since this is inside of a lambda expression, we need to use a type that can be updated atomically, such as `AtomicInteger`, to modify this value. This also permits multiple `INCREMENT` components to run on the same item and stack their effects, like what happens in vanilla.
+
+### `ConditionalEffect`
+Wrapping the type in `ConditionalEffect<?>` allows the enchantment effect component to optionally take effect based on a given [LootContext].
+
+`ConditionalEffect` provides `ConditionalEffect#matches(LootContext context)`, which returns whether the effect should be allowed to run based on its internal `Optional<LootItemConditon>`, and handled serialization and deserialization of its `LootItemCondition`.
+
+Vanilla adds an additional helper method to further streamline the process of checking these conditions: `Enchantment#applyEffects()`. This method takes a `List<ConditionalEffect<T>>`, evaluates the conditions, and runs a `Consumer<T>` on each `T` contained by a `ConditionalEffect` whose condition was met. Since many vanilla enchantment effect components are defined as a `List<ConditionalEffect<?>>`, these can be directly plugged into the helper method like so:
+
+```java
+// `enchant` is an Enchantment instance.
+// `lootContext` is a LootContext instance.
+Enchantment.applyEffects(
+    enchant.getEffects(EnchantmentEffectComponents.KNOCKBACK), // Or whichever other List<ConditionalEffect<T>> you want
+    lootContext,
+    (effectData) -> // Use the effectData (in this example, an EnchantmentValueEffect) however you want.
+);
+```
+
+Registering a custom `ConditionalEffect`-wrapped enchantment effect component type can be done as follows:
+
+```java
+public static final DeferredHolder<DataComponentType<?>, DataComponentType<ConditionalEffect<Increment>>> CONDITIONAL_INCREMENT =
+    ENCHANTMENT_COMPONENT_TYPES.register("conditional_increment",
+        () -> DataComponentType.ConditionalEffect<Increment>builder()
+            .persistent(ConditionalEffect.codec(Increment.CODEC, LootContextParamSets.EMPTY))
```

Comment:
So, for examples, you should never use constants that will never be used by a modder. Some parameter is required for any condition, as all conditions require some context.

I would leave a comment above this line saying that the context provided should include whatever the enchantment should be used for. If you are not sure, use one of the existing `ENCHANTED_*` or as a guideline.